x-pack/plugins/endpoint/public/applications/endpoint/view/hosts/details/components/flyout_sub_header.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+import { EuiFlyoutHeader, CommonProps, EuiButtonEmpty } from '@elastic/eui';
+import styled from 'styled-components';
+
+export type FlyoutSubHeaderProps = CommonProps & {
+  children: React.ReactNode;
+  backButton?: {
+    title: string;
+    onClick: (event: React.MouseEvent) => void;
+    href?: string;
+  };
+};
+
+const StyledEuiFlyoutHeader = styled(EuiFlyoutHeader)`
+  padding: ${props => props.theme.eui.paddingSizes.s};
+
+  &.hasButtons {
+    .buttons {
+      padding-bottom: ${props => props.theme.eui.paddingSizes.s};
+    }
+
+    .back-button-content {
+      padding-left: 0;
+      &-text {
+        margin-left: 0;
+      }
+    }
+  }
+
+  .flyout-content {
+    padding-left: ${props => props.theme.eui.paddingSizes.m};
+  }
+`;
+
+const BUTTON_CONTENT_PROPS = Object.freeze({ className: 'back-button-content' });
```

Comment:
question about why you have to object.freeze this object

Reply:
I am defining this constant at the module level and want to ensure it is never mutated accidentally, since that would impact future instantiations of the component.
Maybe I should have also used a type to set the value as Immutable.